src/game/logic/object/update/laserupdate.cpp

```diff
@@ -0,0 +1,365 @@
+    xfer->xferUnsignedInt(&m_shrinkStartFrame);
+    xfer->xferUnsignedInt(&m_shrinkEndFrame);
+    xfer->xferDrawableID(&m_sourceDrawableID);
+    xfer->xferDrawableID(&m_victimDrawableID);
+    xfer->xferAsciiString(&m_masterBone);
+}
+
+void LaserUpdate::Load_Post_Process()
+{
+    DrawableModule::Load_Post_Process();
+}
+
+ModuleData *LaserUpdate::Friend_New_Module_Data(INI *ini)
+{
+    LaserUpdateModuleData *data = new LaserUpdateModuleData();
+
+    if (ini) {
+        ini->Init_From_INI_Multi_Proc(data, LaserUpdateModuleData::Build_Field_Parse);
+    }
+
+    return data;
+}
+
+Module *LaserUpdate::Friend_New_Module_Instance(Thing *thing, ModuleData const *module_data)
+{
+    return NEW_POOL_OBJ(LaserUpdate, thing, module_data);
+}
+
+float LaserUpdate::Get_Current_Laser_Radius() const
+{
+    const Drawable *drawable = Get_Drawable();
+
+    for (const DrawModule **m = drawable->Get_Draw_Modules(); *m != nullptr; m++) {
+        const LaserDrawInterface *l = (*m)->Get_Laser_Draw_Interface();
+
+        if (l) {
+            return l->Get_Laser_Template_Width() * m_width;
+        }
+    }
+
+    return 0.0f;
+}
+
+void LaserUpdate::Init_Laser(Object const *source_obj,
+    Object const *victim_obj,
+    Coord3D const *start_pos,
+    Coord3D const *end_pos,
+    Utf8String master_bone,
+    int width)
+{
+    const LaserUpdateModuleData *data = Get_Module_Data();
+
+    if (width > 0) {
+        m_grow = true;
+        m_growStartFrame = g_theGameLogic->Get_Frame();
+        m_growEndFrame = width + m_growStartFrame;
+        m_width = 0.0f;
+    } else if (width < 0) {
+        m_shrink = true;
+        m_shrinkStartFrame = g_theGameLogic->Get_Frame();
+        m_shrinkEndFrame = width + m_shrinkStartFrame;
+        m_width = 1.0f;
+    }
+
+    m_masterBone = master_bone;
+
+    if (source_obj) {
+        if (source_obj->Get_Drawable()) {
+            m_sourceDrawableID = source_obj->Get_Drawable()->Get_ID();
+        }
+
+        Update_Start_Pos();
+    } else {
+        if (!start_pos) {
+            g_theGameClient->Destroy_Drawable(Get_Drawable());
+            return;
+        }
+
+        m_startPos = *start_pos;
+    }
+
+    if (victim_obj && !end_pos) {
+        if (victim_obj->Get_Drawable()) {
+            m_victimDrawableID = victim_obj->Get_Drawable()->Get_ID();
+        }
+
+        m_endPos = *victim_obj->Get_Position();
+    } else {
+        if (!end_pos) {
+            g_theGameClient->Destroy_Drawable(Get_Drawable());
+            return;
+        }
+
+        m_endPos = *end_pos;
+    }
+
+    if (!m_muzzleParticleSystemID) {
+        Player *player = g_thePlayerList->Get_Local_Player();
+
+        if (!source_obj || source_obj->Get_Shrouded_Status(player->Get_Player_Index()) <= SHROUDED_PARTIAL) {
+            if (data->m_muzzleParticleSystem.Is_Not_Empty()) {
+                ParticleSystemTemplate *tmplate = g_theParticleSystemManager->Find_Template(data->m_muzzleParticleSystem);

```

Comment:
**tmplate** spelling error.

Reply:
Its not a spelling error, its intentional because template is a keyword in C++.